### PR TITLE
fix(update_user_reports) Batch the snuba calls to avoid query size errors

### PR DESCRIPTION
**Pull request by evanh**
Sometimes this endpoint receives an error from Snuba exceeding the maximum query
size because it has too many event IDs in the where clause. Batch the event IDs
to avoid this errors in the future.

### CODE REVIEW

File: src/sentry/tasks/update_user_reports.py [hunk 1]
```diff
@@ -28,16 +30,23 @@ def update_user_reports(**kwargs):
...
         report_by_event = {r.event_id: r for r in reports}
-        snuba_filter = eventstore.Filter(
-            project_ids=[project_id],
-            event_ids=event_ids,
-            start=now - timedelta(days=2),
-            end=now + timedelta(minutes=5),  # Just to catch clock skew
-        )
-        events = eventstore.get_events(filter=snuba_filter)
+        # Batch these calls to avoid max query size errors
+        chunks = ceil(len(event_ids) / MAX_EVENTS)
```
Comment (wedamija) [inline src/sentry/tasks/update_user_reports.py:38]: Nit: There's a helper for chunking in `sentry.utils.iterators.chunked`

File: src/sentry/tasks/update_user_reports.py [hunk 2]
```diff
@@ -28,16 +30,23 @@ def update_user_reports(**kwargs):
     updated_reports = 0
     samples = None
 
+    MAX_EVENTS = kwargs.get("max_events", 5000)
```
Comment (wedamija) [inline src/sentry/tasks/update_user_reports.py:33]: Is this going to end up too high?

**Review by wedamija — APPROVED** [review]

File: src/sentry/tasks/update_user_reports.py (hunk 2 above)
Comment (evanh) [reply src/sentry/tasks/update_user_reports.py:33]: Some napkin math shows that if the entire query was just event IDs, we could support around 13,000. I figure at this limit (5000) we could in theory have the entire list of event IDs twice and still have plenty of space for the rest of the query.

File: src/sentry/tasks/update_user_reports.py (hunk 1 above)
Comment (evanh) [reply src/sentry/tasks/update_user_reports.py:38]: It's less memory efficient but I changed it.

Comment (wedamija) [reply src/sentry/tasks/update_user_reports.py]: Yeah I noticed that too, we could potentially change the implementation of `chunked` to improve it

**Pull request merged**